src/erlcloud_application_autoscaler.erl

```diff
@@ -318,7 +318,9 @@ delete_scaling_policy(Configuration, BodyConfiguration) ->
                             ScalableDimension :: binary(),
                             ScheduledActionName :: binary(),
                             ServiceNamespace :: binary()
-                        ) -> response().
+                        ) -> {ok, jsx:json_term()} | {error, metadata_not_available
```

Comment:
seems like it would save a lot of copy-pasta to coin an `application_autoscaler_response()` type, no?

Reply:
It would. Do you prefer it?

**Note**: there was already a `response()` type defined, and I didn't want to introduce further ambiguity.

Reply:
Good point, what about an `ok_response()` type to distinguish the response format from the ones that return naked values and raise exceptions on failure?

Reply:
You mean turn some of those `response()`s into `ok_response()` and then have `application_autoscaler_response() :: ok_response() | {error, _}`?

Reply:
I mean something more like 

```erlang
-type ok_response() :: {ok, jsx:json_term()}
                     | {error, metadata_not_available
                             | container_credentials_unavailable
                             | erlcloud_aws:httpc_result_error()}.
```

maybe better to call it `ok_error_response()`? I don't care too much about naming, I'd just prefer to avoid copying the same ok/error response spec everywhere.

Reply:
Sure, I can do that.

Reply:
Fixed in 7b52338.